src/main/java/com/cottonfactory/products/services/PantsService.java

```diff
@@ -0,0 +1,29 @@
+package com.cottonfactory.products.services;
+
+import com.cottonfactory.products.entities.Pants;
+import com.cottonfactory.products.repositories.PantsRepository;
+import com.cottonfactory.products.services.dtos.PantsDto;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class PantsService {
+
+    public PantsService(PantsRepository pantsRepository) {
+        this.pantsRepository = pantsRepository;
+    }
+
+    private PantsRepository pantsRepository;
+
+    public List<PantsDto> findAllPants() {
+        return pantsRepository.findAll().stream().map(pants -> {
+            PantsDto pantsDto =

```

Comment:
This is a suggestion, could try BeanUtils.copyProperties as well, in case there is a large amount of fields and names repeat often between Entity and DTO.

Reply:
The reviewer said we can use this in subsequent iterations or if we have a large number of fields.